src/bot/integrations/discord.ts

```diff
@@ -61,27 +61,37 @@ class Discord extends Integration {
   }, 'general')
   cannotJoinToServerBtn = null;
 
+  @settings('bot')
+  @ui({
+    type: 'discord-guild',
+  })
+  guild = '';
+
   @settings('bot')
   @ui({
     type: 'discord-channel',
+    if: () => self.guild.length === 0 || self.guild.length === 0,
   })
   listenAtChannels = '';
 
   @settings('bot')
   @ui({
     type: 'discord-channel',
+    if: () => self.guild.length === 0 || self.guild.length === 0,
   })
   sendOnlineAnnounceToChannel = '';
 
   @settings('bot')
   @ui({
     type: 'discord-channel',
+    if: () => self.guild.length === 0 || self.guild.length === 0,
   })
   sendGeneralAnnounceToChannel = '';
 
   @settings('mapping')
   @ui({
     type: 'discord-mapping',
+    if: () => self.guild.length === 0 || self.guild.length === 0,
```

Comment:
shouldn't this be

```suggestion
    if: () => self.guild.length > 0,
```

Reply:
Also you can probably add alertbox for if: () => self.guild.length === 0 to inform that there are more settings after guild is set?